contracts/contracts/vault/OETHVaultCore.sol

```diff
@@ -145,4 +157,230 @@ contract OETHVaultCore is VaultCore {
+    function addWithdrawalQueueLiquidity() external {
+        // Stream any harvested rewards (WETH) that are available to the Vault
+        IDripper(dripper).collect();
+
+        _addWithdrawalQueueLiquidity();
+    }
+
+    /// @dev Adds WETH to the withdrawal queue if there is a funding shortfall.
+    /// This assumes 1 WETH equal 1 OETH.
+    function _addWithdrawalQueueLiquidity()
+        internal
+        returns (uint256 addedClaimable)
+    {
+        WithdrawalQueueMetadata memory queue = withdrawalQueueMetadata;
+
+        // Check if the claimable WETH is less than the queued amount
+        uint256 queueShortfall = queue.queued - queue.claimable;
+
+        // No need to get WETH balance if all the withdrawal requests are claimable
+        if (queueShortfall > 0) {
+            uint256 wethBalance = IERC20(weth).balanceOf(address(this));
+
+            // Of the claimable withdrawal requests, how much is unclaimed?
+            uint256 unclaimed = queue.claimable - queue.claimed;
+            if (wethBalance > unclaimed) {
+                uint256 unallocatedWeth = wethBalance - unclaimed;
+
+                // the new claimable amount is the smaller of the queue shortfall or unallocated weth
+                addedClaimable = queueShortfall < unallocatedWeth
+                    ? queueShortfall
+                    : unallocatedWeth;
+                uint256 newClaimable = queue.claimable + addedClaimable;
+
+                // Store the new claimable amount back to storage
+                withdrawalQueueMetadata.claimable = uint128(newClaimable);
+
+                // emit a WithdrawalClaimable event
+                emit WithdrawalClaimable(newClaimable, addedClaimable);
+            }
+        }
+    }
+
+    /***************************************
+                View Functions
+    ****************************************/
+
+    function _wethAvailable() internal view returns (uint256 wethAvailable) {
+        WithdrawalQueueMetadata memory queue = withdrawalQueueMetadata;
+
+        // Check if the claimable WETH is less than the queued amount
+        uint256 queueShortfall = queue.queued - queue.claimable;
+        uint256 wethBalance = IERC20(weth).balanceOf(address(this));
+        // Of the claimable withdrawal requests, how much is unclaimed?
+        uint256 unclaimed = queue.claimable - queue.claimed;
+
+        if (wethBalance > queueShortfall + unclaimed) {
+            wethAvailable = wethBalance - queueShortfall - unclaimed;
+        }
+    }
+
+    function _checkBalance(address _asset)
+        internal
+        view
+        override
+        returns (uint256 balance)
+    {
+        balance = super._checkBalance(_asset);
+
+        if (_asset == weth) {
+            WithdrawalQueueMetadata memory queue = withdrawalQueueMetadata;
+            // Need to remove WETH that is reserved for the withdrawal queue
+            return balance + queue.claimed - queue.queued;
```

Comment:
🔴 This can cause the Vault to break due to underflow issues.

The code assumes that `balance + queue.claimed` will be always be greater than `queue.queued`. However, that's not the case all the time.

Someone can come and do a `requestWithdrawal` for an amount that's larger than whatever WETH Vault holds and essentially break all of our contracts

Reply:
This has been fixed in the new `_wethAvailable` implementation